spec/requests/file_download_spec.rb

```diff
@@ -31,50 +28,110 @@
     allow(Aws::S3::Client).to receive(:new).and_return(s3)
   end
 
-  context 'when file exists' do
-    before do
-      s3.stub_responses(:head_object, {})
-      s3.stub_responses(:get_object, { body: file_fixture('encrypted_file').read })
+  context 'downloading a file' do
+    let(:do_get!) do
+      get "/service/service-slug/user/#{user_id}/28d-fingerprint", headers: headers
     end
 
-    it 'returns status 200' do
-      do_get!
-      expect(response).to be_successful
+    context 'when file exists' do
+      before do
+        s3.stub_responses(:head_object, {})
+        s3.stub_responses(:get_object, { body: file_fixture('encrypted_file').read })
+      end
+
+      it 'returns status 200' do
+        do_get!
+        expect(response).to be_successful
+      end
+
+      it 'returns file' do
+        do_get!
+        expect(response.body).to eql("lorem ipsum\n")
+      end
+    end
+
+    context 'when file does not exist' do
+      before do
+        s3.stub_responses(:head_object, 'NotFound')
+      end
+
+      it 'returns status 404' do
+        do_get!
+        expect(response).to be_not_found
+      end
+
+      it 'returns correct json' do
+        do_get!
+        hash = JSON.parse(response.body)
+        expect(hash['code']).to eql(404)
+        expect(hash['name']).to eql('not-found')
+      end
     end
 
-    it 'returns file' do
-      do_get!
-      expect(response.body).to eql("lorem ipsum\n")
+    context 'when there is a problem' do
+      it 'returns 503' do
+        downloader = double('downloader', exists?: true)
+        allow(Storage::S3::Downloader).to receive(:new).and_return(downloader)
+        allow(downloader).to receive(:contents).and_raise(StandardError.new)
+
+        do_get!
+
+        expect(response.status).to eql(503)
+      end
```

Comment:
When this is raised what happens with the Job? Just failed and that failed doesn't cascade with other jobs for the same submission?

Reply:
So by the time it gets here the job that would fail in the submitter would be for a single email and should not affect any other jobs 🤞